components/web_discovery/browser/value_transform_unittest.cc

```diff
@@ -61,6 +108,37 @@ TEST(WebDiscoveryValueTransformTest, FilterExactTransform) {
   EXPECT_FALSE(invalid_transform);
 }
 
+TEST(WebDiscoveryValueTransformTest, RemoveParamsTransform) {
+  // Test removeParams
+  auto transform = CreateTransform(
+      R"(["removeParams", ["utm_source", "utm_medium", "fbclid"]])");
+  ASSERT_TRUE(transform);
+
+  // Test URL with params to remove
+  auto result = transform->Process(
+      "https://"
+      "example.com?utm_source=google&param1=value1&utm_medium=cpc&param2="
+      "value2");
+  EXPECT_EQ(result, "https://example.com/?param1=value1&param2=value2");
+
+  // Test URL with no query params
+  result = transform->Process("https://example.com");
+  EXPECT_EQ(result, "https://example.com");
+
+  // Test URL with all params removed
+  result = transform->Process(
+      "https://example.com?utm_source=google&utm_medium=cpc");
+  EXPECT_EQ(result, "https://example.com/");
+
+  // Test invalid URL
+  result = transform->Process("not-a-url");
+  EXPECT_EQ(result, std::nullopt);
+
+  // Test URL with no matching params
```

Comment:
I'd suggest another one just to emphasize that we're not making assumptions about the format of the query string in this function:
```
https://example.com?utm_source=google;utm_medium=cpc;utm_content=something
```
(should be untouched)